src/mpid/ch4/src/ch4_impl.h

```diff
@@ -968,151 +968,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_compute_acc_op(void *source_buf, int source_
-
-        MPI_Aint actual_iov_len, actual_iov_bytes;
-        MPIR_Typerep_to_iov(NULL, target_count, target_dtp, 0, typerep_vec, vec_len,
-                            source_count * source_dtp_size, &actual_iov_len, &actual_iov_bytes);
-        vec_len = actual_iov_len;
-
-        type = dtp->basic_type;
-        MPIR_Assert(type != MPI_DATATYPE_NULL);
-
-        MPIR_Assert(type == source_dtp);
-        type_size = source_dtp_size;
-        type_extent = source_dtp_extent;
-        /* If the source buffer has been packed by the caller, the distance between
-         * two elements can be smaller than extent. E.g., predefined pairtype may
-         * have larger extent than size.*/
-        /* when predefined pairtype have larger extent than size, we'll end up
-         * missaligned access. Memcpy the source to workaround the alignment issue.
-         */
-        char *src_ptr = NULL;
-        if (src_kind == MPIDIG_ACC_SRCBUF_PACKED) {
-            src_type_stride = source_dtp_size;
-            if (source_dtp_size < source_dtp_extent) {
-                src_ptr = MPL_malloc(source_dtp_extent, MPL_MEM_OTHER);
-            }
-        } else {
-            src_type_stride = source_dtp_extent;
-        }
-
-        i = 0;
-        curr_loc = typerep_vec[0].iov_base;
-        curr_len = typerep_vec[0].iov_len;
-        accumulated_count = 0;
-        while (i != vec_len) {
-            if (curr_len < type_size) {
-                MPIR_Assert(i != vec_len);
-                i++;
-                curr_len += typerep_vec[i].iov_len;
-                continue;
-            }
-
-            MPI_Aint count;
-            MPIR_Assign_trunc(count, curr_len / type_size, MPI_Aint);
-
-            if (src_ptr) {
-                MPI_Aint unpacked_size;
-                MPIR_Typerep_unpack((char *) source_buf + src_type_stride * accumulated_count,
-                                    source_dtp_size, src_ptr, 1, source_dtp, 0, &unpacked_size);
-                (*uop) (src_ptr, (char *) target_buf + MPIR_Ptr_to_aint(curr_loc), &count, &type);
-            } else {
-                (*uop) ((char *) source_buf + src_type_stride * accumulated_count,
-                        (char *) target_buf + MPIR_Ptr_to_aint(curr_loc), &count, &type);
-            }
-
-            if (curr_len % type_size == 0) {
-                i++;
-                if (i != vec_len) {
-                    curr_loc = typerep_vec[i].iov_base;
-                    curr_len = typerep_vec[i].iov_len;
-                }
-            } else {
-                curr_loc = (void *) ((char *) curr_loc + type_extent * count);
-                curr_len -= type_size * count;
-            }
-
-            accumulated_count += count;
-        }
-
-        MPL_free(src_ptr);
-        MPL_free(typerep_vec);
-    }
+    mpi_errno =
+        MPIR_Typerep_rma_reduce(source_buf, source_count, source_dtp, target_buf, target_count,
+                                target_dtp, acc_op, src_kind);
```

Comment:
I think `MPIR_Typerep_op` is a better name. It describes what it does -- adding source_buf onto target_buf. Whether it is used in RMA or other places is irrelevant.

Reply:
I have changed the name to `MPIR_Typerep_op`.